cpp/src/io/json/json_column.cu

```diff
@@ -411,11 +411,33 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> device_json_co
       std::vector<column_name_info> column_names{};
       size_type num_rows{json_col.num_rows};
       // Create children columns
-      for (auto const& col_name : json_col.column_order) {
-        auto const& col = json_col.child_columns.find(col_name);
-        column_names.emplace_back(col->first);
-        auto& child_col           = col->second;
+      auto const& col_order = prune_columns and schema.has_value() and
+                                  schema.value().column_order.has_value() and
+                                  not schema.value().column_order->empty()
```

Comment:
I don't know if this works and/or is any better; nested optional checking looks so clunky to me.
```suggestion
      auto const& col_order = prune_columns and not schema.value_or({}).column_order.value_or({}).empty()
```

Reply:
`value_or({})` does not work. so, 
The code will be like
`not schema.value_or(schema_element{}).column_order.value_or(std::vector<std::string>{}).empty();`